test/extended/apiserver/resiliency.go

```diff
@@ -0,0 +1,65 @@
+package apiserver
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	"github.com/openshift/origin/test/extended/util/disruption"
+	"github.com/openshift/origin/test/extended/util/disruption/controlplane"
+	types "k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/test/e2e/upgrades"
+)
+
+func forceApiRollout(oc *exutil.CLI) {
+	redeploymentReason := fmt.Sprintf(`{"spec":{"forceRedeploymentReason":"resiliency-test-%v"}}`, uuid.NewUUID())
+
+	_, err := oc.AdminOperatorClient().OperatorV1().KubeAPIServers().Patch(context.Background(), "cluster", types.MergePatchType,
+		[]byte(redeploymentReason), metav1.PatchOptions{})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+}
+
+func isApiUp(oc *exutil.CLI) (bool, error) {
+	_, err := oc.AdminKubeClient().CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
```

Comment:
this is not a good check. Check readiness. This answers success far too early.

Reply:
fixed